stan/math/prim/mat/fun/matrix_exp.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_MATRIX_EXP_HPP
+#define STAN_MATH_PRIM_MAT_FUN_MATRIX_EXP_HPP
+
+#include <unsupported/Eigen/MatrixFunctions>
+#include <stan/math/prim/mat/fun/MatrixExponential.h>
+#include <stan/math/prim/mat/fun/matrix_exp_2x2.hpp>
+
+namespace stan {
+    namespace math {
+
+        /**
+         * Return the matrix exponential of the input
+         * matrix. 
+         *
+         * @param[in] A A matrix
+         * @return Matrix exponential. 
+         */
+
```

Comment:
No blank line between doxygen doc and start of function.  Also, you need `@tparam T Scalar type.  And probalby not `A A matrix` but `A Matrix to exponentiate.`